internal/server/data/data_test.go

```diff
@@ -51,24 +60,69 @@ func setupLogging(t *testing.T) {
 	})
 }
 
-func TestSnowflakeIDSerialization(t *testing.T) {
-	db := setup(t)
+// dbDrivers returns the list of database drivers to test.
+// Set POSTGRESQL_CONNECTION to a postgresql connection string to run tests
+// against postgresql.
+func dbDrivers(t *testing.T) []gorm.Dialector {
+	t.Helper()
+	var drivers []gorm.Dialector
+
+	sqlite, err := NewSQLiteDriver("file::memory:")
+	assert.NilError(t, err, "sqlite driver")
+	drivers = append(drivers, sqlite)
+
+	pgConn, ok := os.LookupEnv("POSTGRESQL_CONNECTION")
+	switch {
+	case ok:
+		pgsql, err := NewPostgresDriver(pgConn)
+		assert.NilError(t, err, "postgresql driver")
+
+		db, err := gorm.Open(pgsql)
+		assert.NilError(t, err, "connect to postgresql")
+		t.Cleanup(func() {
+			assert.NilError(t, db.Exec("DROP SCHEMA testing CASCADE").Error)
+		})
+		assert.NilError(t, db.Exec("CREATE SCHEMA testing").Error)
+
+		pgsql, err = NewPostgresDriver(pgConn + " search_path=testing")
+		assert.NilError(t, err, "postgresql driver")
+
+		drivers = append(drivers, pgsql)
+	case os.Getenv("CI") != "":
+		t.Fatalf("CI must test all drivers, set POSTGRESQL_CONNECTION")
```

Comment:
This is to ensure we always run everything in CI.

Example failure when CI is not setup properly: https://github.com/infrahq/infra/runs/6360878148?check_suite_focus=true